themes/default/content/docs/concepts/inputs-outputs/_index.md

```diff
@@ -13,30 +13,346 @@ aliases:
+{{< chooser language "javascript,typescript,python,go,csharp,java" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+// TODO
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const myId = new random.RandomId("mine", {
+    byteLength: 8, // byte_length is an input
+});
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+my_id = random.RandomId("mine",
+    byte_length=8 # byte_length is an input
+)
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+
+myId, err := random.NewRandomId(ctx, "mine", &random.RandomIdArgs{
+	ByteLength: pulumi.Int(8), // ByteLength is an input
+})
+if err != nil {
+	return err
+}
+
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var myId = new Random.RandomId("mine", new()
+{
+    ByteLength = 8, // ByteLength is an input
+});
+
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+var myId = new RandomId("mine", RandomIdArgs.builder()
+    .byteLength(8) // byteLength is an input
+    .build());
+
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+_Inputs_ are generally representations of the parameters to the underlying API call of any resource that Pulumi is managing.
+
+The simplest way to create a resource with its required _inputs_ is to use a _plain value_.
+
+{{< chooser language "javascript,typescript,python,go,csharp,java" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+const key = new tls.PrivateKey("my-private-key", {
+    algorithm: "ECDSA", // ECDSA is a plain value
+});
+
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const key = new tls.PrivateKey("my-private-key", {
+    algorithm: "ECDSA", // ECDSA is a plain value
+});
+
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+key = tls.PrivateKey("my-private-key",
+  algorithm="ECDSA", # ECDSA is a plain value
+)
+
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+
+key, err := tls.NewPrivateKey(ctx, "my-private-key", &tls.PrivateKeyArgs{
+	Algorithm:  pulumi.String("ECDSA"), // ECDSA is a plain value
+})
+if err != nil {
+	return err
+}
+
+
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var key = new PrivateKey("my-private-key", new PrivateKeyArgs{
+    Algorithm = "ECDSA", // ECDSA is a plain value
+});
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+// TODO
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+{{% notes %}}
+_Plain value_ in this document is used to describe a standard string, boolean, integer or other typed value or data structure in your language of choice. _Plain_ value_ is a way of differentiating these language specific values from Pulumi's asynchronous values.
```

Comment:
I don’t think we should introduce the term “asynchronous values” here. Can’t we just call them “outputs”, which is the name they are referred as everywhere else?

Reply:
I've removed references to asynchronous